stdlib/http/src/main/java/org/ballerinalang/net/http/ClientListener.java

```diff
@@ -0,0 +1,133 @@
+
+    public void setConnectionInfo(WebSocketOpenConnectionInfo connectionInfo) {
+        this.connectionInfo = connectionInfo;
+    }
+
+    @Override
+    public void onHandshake(WebSocketHandshaker webSocketHandshaker) {
+        throw new WebSocketException("onHandshake and onOpen is not supported for WebSocket client service");
+    }
+
+    @Override
+    public void onMessage(WebSocketTextMessage webSocketTextMessage) {
+        try {
+            WebSocketDispatcher.dispatchTextMessage(connectionInfo, webSocketTextMessage);
+        } catch (IllegalAccessException e) {
+            // Ignore as it is not possible have an Illegal access
+        }
+    }
+
+    @Override
+    public void onMessage(WebSocketBinaryMessage webSocketBinaryMessage) {
+        try {
+            WebSocketDispatcher.dispatchBinaryMessage(connectionInfo, webSocketBinaryMessage);
+        } catch (IllegalAccessException e) {
+            // Ignore as it is not possible have an Illegal access
+        }
+    }
+
+    @Override
+    public void onMessage(WebSocketControlMessage webSocketControlMessage) {
+        try {
+            WebSocketDispatcher.dispatchControlMessage(connectionInfo, webSocketControlMessage);
+        } catch (IllegalAccessException e) {
+            // Ignore as it is not possible have an Illegal access
+        }
+    }
+
+    @Override
+    public void onMessage(WebSocketCloseMessage webSocketCloseMessage) {
+        ObjectValue webSocketClient = connectionInfo.getWebSocketEndpoint();
+        int statusCode = webSocketCloseMessage.getCloseCode();
+        if (hasRetryConfig(webSocketClient) && statusCode == STATUS_CODE_ABNORMAL_CLOSURE &&
+                reconnect(connectionInfo)) {
+            return;
+        } else {
+            if (hasRetryConfig(webSocketClient) && statusCode != STATUS_CODE_ABNORMAL_CLOSURE) {
+                logger.info("Couldn't connect to the server because the given server: " +
+                        webSocketClient.getStringValue(WebSocketConstants.CLIENT_URL_CONFIG) +
+                        "have sent the close request to the client.");
+            }
+        }
+        setCloseMessage(connectionInfo, webSocketCloseMessage);
+    }
+
+    @Override
+    public void onError(WebSocketConnection webSocketConnection, Throwable throwable) {
+        ObjectValue webSocketClient = connectionInfo.getWebSocketEndpoint();
+        if (hasRetryConfig(webSocketClient)) {
+            if (throwable instanceof IOException && reconnect(connectionInfo)) {
+                return;
+            } else {
+                logger.info("Unable to do the retry because the connection: " +
+                        webSocketClient.getStringValue(WebSocketConstants.CLIENT_URL_CONFIG) +
+                        "has some issue that needs to fix.");
+            }
+        }
+        WebSocketDispatcher.dispatchError(connectionInfo, throwable);
```

Comment:
I think we are duplicating code in HandShakeListener `onError` and here? Shall we remove duplication by using a util method if possible?